### PR TITLE
CMake Documentation

### PR DESCRIPTION
**Pull request by stxticOVFL**
Please thoroughly review the new documentation, ensuring it is clear, correct, and easy to follow. Please approve or request changes if there is anything to be done.

### CODE REVIEW

**Review by Sonic-Geared — APPROVED** [review]
honestly, this was a huge improvement from the old cmake documentation, now the users won't have to be getting off a page to go to another page and vice-versa, the addition of the video filters in the FAQ was also a nice addition to the new documentation.
so in my opinion, the new documentation became more improved than before, and this is why i approve those changes, even if I'm someone that has no involvement with none of RDC's decomps, i feel the necessity to approve this to make the cmake documentation more easy & better to understand.

**Comment by MegAmi24** [comment]
I get CMake is the preferred method and all, but I think keeping the instructions for the Visual Studio solution would be more beneficial than taking it out completely.

**Review by jvalecillos — COMMENTED** [review]
I tried to compile this today and I had to do some fixes. Here are my suggested changes.

~I still didn't manage to make it work with my version of Sonic Mania though, maybe I'm missing something else.~ I was confused about how to compile it and use it for Sonic Mania, and then I noticed that the executable`RSDKv5` and the `Game.so` (from [Sonic Mania Decompilation](https://github.com/Rubberduckycooly/Sonic-Mania-Decompilation)) are compiled separately, but you need both to run the game.

**Pull request merged**
